### DISCUSSION TITLE
content_available does not work to trigger data-only notifications in ios

### DISCUSSION THREAD

**Discussion by nick-burczyk-cs** [invertase/react-native-firebase#7219] [General]
**Disclaimer: This is very similar to several posts, but I have been through all of them and the solutions given did not work for me.**

Previously, I was using native push notifications and react-native-firebase for handling, and everything was working perfectly. I added Notifee and switched to data-only messages, and now iOS is throwing/triggering nothing. It is like the messages are not arriving. 

## JSON Payload
```json
{
    "to" : "FCM_TOKEN",
    "content_available":true,
    "apns": {        // <=== This object is did nothing, but was borrowed from [this post](https://github.com/invertase/react-native-firebase/issues/1875#issue-405198789)
        "payload": {
          "aps": {
            "alert": {
              "title":"hello ios",
              "body":"world ios"
            }
          }
        }
    },
    "priority":"high",
    "data":{ 
        "notifee": {
            "title":"Hello",
            "body":"World",
            "data": { "name": "Nick" }
        }
    }
}
```
## Implementation in `useNotifications` hook
```js
// Foreground handler
const handleIncoming = async (remoteMessage: RemoteMessage) => {
    console.log('=== Incoming ===', remoteMessage);    // <=== Does not fire
    // ... Do more things (irrelevant, does not reach this)
};

// Background handler
const handleTapToOpenNOTIFEE = async (event: Event) => {
    console.log('=== EVENT ===', JSON.stringify(event)) // <=== Does not fire
    // ...Do more things (irrelevant, does not reach this)
};

useEffect(() => {
      // handle when app is open in foreground
      messaging().onMessage(handleIncoming);

      // handle when app is running in background
      notifee.onBackgroundEvent(handleTapToOpenNOTIFEE);
  }, [allowPushNotifications, hasAccessToNoticeBoard]);
```
## Implimentation in `index.js`
```js
messaging().setBackgroundMessageHandler(async (remoteMessage) => {
  console.log('remoteMessage in BG handler', remoteMessage); // <=== does not fire
  await notifee.requestPermission();
    const channelId = await notifee.createChannel({
      // ... android channel props
  });

  const parsed = JSON.parse(remoteMessage.data.notifee || '');

  await notifee.displayNotification({ // <--- never reaches this
    title: `${parsed?.title}`,
    body: `${parsed?.body}`,
    data: parsed.data,
    android: {
      channelId,
      color: AndroidColor.LIME,
      smallIcon: 'ic_small_icon',
      pressAction: {
        id: 'default',
      },
    },
  });
});
```

## Fixes attempted
* Adding `content_available`, `content-available`, and `contentAvailable` in the payload.
* Adding APNS object in payload
* Adding headers: `{ apns-push-type: background, apns-priority: 5, apns-topic: <DOMAIN_NAME> }`
* removing `priority: true` from the payload (per a SO suggestion)

## Package versions
`react-native`: 0.64.2
`@notifee/react-native`: 7.8.0,
`@react-native-firebase/messaging`: 15.4.0

## Conclusion
Discerning the device type from the backend and adjusting the payload is obviously not an option, but exploring that is almost sounding attractive 😂.

As stated, I have gone through every single similar issue with no success. 

Thanks for any help you are able to give.

**Comment by UsamaUthman** [comment]
Try it on reel device, not simulator 
and just make it like this 
`
 {
            to: token,
            data: {
              json: JSON.stringify(messageData),
            },
            content_available: true,
            priority: "high",
          };
`
it works on reel device.
